Prevent gcc from linking with libpthread if -pthread[s] is passed as an option.
libpthread is a filter in illumos; all functionality has been migrated to libc.

Some build systems such as meson/ninja pass this option; glib2 now requires
meson/ninja to build. Since glib2 is linked with hald in gate, the spurious
libpthread causes build warnings.

+++ a~/gcc/config/sol2.h	1970-01-01 00:00:00
+++ a/gcc/config/sol2.h	1970-01-01 00:00:00
@@ -157,7 +157,6 @@
 #undef LIB_SPEC
 #define LIB_SPEC \
   "%{!symbolic:\
-     %{pthreads|pthread:-lpthread} \
      %{p|pg:-ldl} -lc}"
 
 #ifndef CROSS_DIRECTORY_STRUCTURE
